### ISSUE TITLE
every db connection causes UPDATE

### ISSUE THREAD

**Issue by cdpb** [arnowelzel/lightbox-photoswipe#37]
I am digging in a performance issue with my mariadb lately and found this. It seems that your plugin is performing a UPDATE query along side every (? can't verify yet) regular UPDATE or at least every new connection coming in. I am not familiar with the wp_options table, ~~but as far as my log goes, only the photoswipe plugin and the wp_statistics use it - out of a bunch of plugins.~~
Anyway. Can you confirm this is your intendend behaviour? ~~And if so, is it really necessary to store the db_version, something really static, in a field which gets overwritten every couple of seconds?~~

EDIT: google helps with wp_options. Okay, I get it now, the value does not get overwritten every couple of seconds - but it gets reset to the same value every couple of seconds. The question is still, is this your intendend behaviour or something to fix.

Thanks for your work anyways :+1: 

	          3551 Query	UPDATE `wp_options` SET `option_value` = '24' WHERE `option_name` = 'lightbox_photoswipe_db_version'
		  3552 Query	UPDATE `wp_options` SET `option_value` = '24' WHERE `option_name` = 'lightbox_photoswipe_db_version'
		  3553 Query	UPDATE `wp_options` SET `option_value` = '24' WHERE `option_name` = 'lightbox_photoswipe_db_version'
		  3553 Query	UPDATE `wp_options` SET `option_value` = '1590480084' WHERE `option_name` = 'wp_statistics_check_useronline'
		  3554 Query	UPDATE `wp_options` SET `option_value` = '24' WHERE `option_name` = 'lightbox_photoswipe_db_version'
		  3554 Query	UPDATE `wp_options` SET `option_value` = '1590480085' WHERE `option_name` = 'wp_statistics_check_useronline'

**Comment by arnowelzel** [comment]
Yes it is neccessary to store the version to decide which updates in settings or database structure are needed after an update. Also see https://github.com/arnowelzel/lightbox-photoswipe/blob/2.100/lightbox-photoswipe.php#L1172

I was not aware, that `update_option()` does not check first if an update is really needed, if the value does not change. So far there was never a performance issue with that - even on sites with more than 10000 visits per hour. However I understand the problem and will change that for the next update:

```
if (intval($db_version) < 25) {
    update_option('lightbox_photoswipe_db_version', 25);
}
```

**Comment by cdpb** [comment]
I couldn't even edit my OT, wow that was a fast reply.
Thanks again for your work and thank you for the fix!

**Comment by arnowelzel** [comment]
Fixed with 3e297abbcaafac231bfa3f9a81ec0e441a26250b
